docs/proposals/20191030-machine-health-checking.md

```diff
@@ -167,7 +164,7 @@ type target struct {
 - Annotates unhealthy target machines as described above.
 
 Out of band:
-- The owning controller observes the annotation and is responsible to remediate the machine.
+- The owning controller observes the condition and is responsible to remediate the machine.
```

Comment:
This shouldn't be true anymore, it should be a different condition to signal remediation

Reply:
We discussed about having `RemediationByOwnerSucceeded` or `OwnerRemediationSucceded` for example which is a contract that a Machine defines (and own), but other controllers have to respect.

Reply:
Can you explain why we need more than one condition?

Reply:
Machine type should define the healthy annotation to determine if it's healthy or not, but any sort of remediation should probably go in a different condition, specific to which strategy should be taken.

For example:
- `OwnerRemediationSucceded` is applied when remediation is needed and we expect the controller-owner to remediate the Machine.
- `ExternalRemediationSucceeded` is instead applied when remediation is expected to happen outside, and the owning controllers shouldn't take any actions to remediate.

These would be defined in the Machine type, but the contract would state that they would be applied by a health check controller (MHC in this case).

cc @JoelSpeed (pls ping others from External Remediation as well)

Reply:
cc @beekhof @n1r1

Reply:
Can we punt the details of external remediation to that proposal? I don't see why we need to resolve that here

Reply:
We can, but we should at least specify OwnerRemediationSucceded

Reply:
This has a lot of overlap with the external remediation proposal. For example, if this condition approach is a suitable signalling mechanism, then the external remediation CRDs (or at least MHC creating them from templates) doesn't add that much value

Why could KCP not use the external remediation proposal to create a `KCPMachineRemediation` CRD to act as the signal for requiring remediation?

This is kind of an aside, but I'm not completely familiar with the actions that KCP needs to perform before deleting the machine and I'd like to understand this a bit better, does anyone know if this is documented somewhere?

Reply:
Yeah, that's why I'm trying to push that discussion out of this issue. This is just an update to the annotation proposal that uses conditions rather than annotations.

Reply:
> We can discuss separately whether conditions vs CRDs make sense for external remediation.

Personally I think we should be having that discussion before we do the KCP remediation work.

The concern I have is that MHC is becoming complicated with lots of different paths to various remediations that might trample one another

Ie, if it's a machine it gets an annotation, unless theres an object reference to a template, but if it's a KCP machine, it gets an annotation even if there is an object reference to a template? 

I can see this becoming very confusing and would prefer that we have a discussion about how we want this to look long term. I don't mind baby steps to get there, I'm in favour of that, but at the moment, I don't think we have a long term goal or vision and I'm concerned that we might cause more problems if we implement stuff without taking a step back and looking at it.

I don't want to block the effort of getting KCP remediation working, I can see conditions working as an intermediary, but I want to have the conversation about aligning, long term, either external needs to be triggered by conditions too, or KCP needs to be CRD based, or they both need to change